pkg/util/informermanager/informermanager.go

```diff
@@ -200,13 +206,18 @@ func (m *informerManager) processFTC(
 		ctx, cancel := context.WithCancel(ctx)
 		go informer.Informer().Run(ctx.Done())
 
-		m.lastObservedFTCs[ftcName] = ftc
 		m.informers[ftcName] = informer
 		m.informerCancelFuncs[ftcName] = cancel
 		m.eventHandlerRegistrations[ftcName] = map[*EventHandlerGenerator]cache.ResourceEventHandlerRegistration{}
 		m.lastAppliedFTCsCache[ftcName] = map[*EventHandlerGenerator]*fedcorev1a1.FederatedTypeConfig{}
 	}
 
+	for _, handler := range m.ftcUpdateHandlers {
+		handler(m.lastObservedFTCs[ftcName], ftc)
```

Comment:
Do we need asynchronous processing here?

Reply:
I don't think it is necessary for now.